sql-plugin/pom.xml

```diff
@@ -91,11 +84,70 @@
     </dependencies>
 
     <profiles>
+        <profile>
+            <id>default</id>
+            <activation>
+                <activeByDefault>true</activeByDefault>
+            </activation>
+            <dependencies>
+                <dependency>
+                    <groupId>org.apache.spark</groupId>
+                    <artifactId>spark-hive_${scala.binary.version}</artifactId>
+                </dependency>
+                <dependency>
+                    <groupId>org.apache.spark</groupId>
+                    <artifactId>spark-sql_${scala.binary.version}</artifactId>
+                </dependency>
+            </dependencies>
+        </profile>
+        <profile>
+            <id>release311cdh</id>
+            <activation>
+                <property>
+                    <name>buildver</name>
+                    <value>311cdh</value>
+                </property>
+            </activation>
+            <dependencies>
+                <dependency>
+                    <groupId>org.apache.spark</groupId>
+                    <artifactId>spark-sql_${scala.binary.version}</artifactId>
+                    <version>${spark311cdh.version}</version>
+                    <exclusions>
+                        <exclusion>
+                            <groupId>org.apache.curator</groupId>
+                            <artifactId>curator-recipes</artifactId>
+                        </exclusion>
+                    </exclusions>
+                    <scope>provided</scope>
+                </dependency>
+                <dependency>
+                    <groupId>org.apache.spark</groupId>
+                    <artifactId>spark-hive_${scala.binary.version}</artifactId>
+                    <version>${spark311cdh.version}</version>
+                </dependency>
+                <dependency>
+                    <groupId>org.apache.curator</groupId>
+                    <artifactId>curator-recipes</artifactId>
+                    <version>4.3.0.7.2.7.0-184</version>
```

Comment:
By pulling in a different dependency version here, is there any concern with conflicts? Does it need to be shaded as we have done in other cases where we pull in a different dependency version? This comment applies to other places where the curator-recipes dependency was overridden as well.

Reply:
good question, I started just copying what they did in the shim layer so I'll check on this.

Reply:
I went ahead and changed all these to be provided as cdh environment should have these. 
 I also added a task for this in https://github.com/NVIDIA/spark-rapids/issues/3223 for us to check when we start combining these to make sure it doesn't leak/conflict.

Reply:
I also checked the final dist jar for cdh to make sure nothing else was in there and its good now